test/unit/processing/confirm-due-dates.test.js

```diff
@@ -0,0 +1,588 @@
+const { AP, AR } = require('../../../app/ledgers')
+const confirmDueDates = require('../../../app/processing/confirm-due-dates')
+const moment = require('moment')
+
+describe('confirm due dates', () => {
+  test('should return no change if no previous requests', () => {
+    const paymentRequests = [{
+      ledger: AP,
+      dueDate: '09/11/2022',
+      schedule: 'Q4',
+      value: -100
+    }]
+    const confirmedPaymentRequests = confirmDueDates(paymentRequests)
+    expect(confirmedPaymentRequests[0].dueDate).toBe('09/11/2022')
+    expect(confirmedPaymentRequests[0].schedule).toBe('Q4')
+  })
+
+  test('should return no change if no previous requests as empty array', () => {
+    const paymentRequests = [{
+      dueDate: '09/11/2022',
+      schedule: 'Q4',
+      value: -100
+    }]
+    const confirmedPaymentRequests = confirmDueDates(paymentRequests, [])
+    expect(confirmedPaymentRequests[0].dueDate).toBe('09/11/2022')
+    expect(confirmedPaymentRequests[0].schedule).toBe('Q4')
+  })
+
+  test('should not update schedule when quarterly and no outstanding payments', () => {
+    const paymentRequests = [{
+      ledger: AP,
+      dueDate: '09/11/2022',
+      schedule: 'Q4',
+      value: -100
+    }]
+    const previousPaymentRequests = [{
+      paymentRequestNumber: 1,
+      ledger: AP,
+      dueDate: moment(new Date()).subtract(2, 'year').format('DD/MM/YYYY'),
+      schedule: 'Q4',
+      value: 1000,
+      settledValue: 1000
+    }]
+    const confirmedPaymentRequests = confirmDueDates(paymentRequests, previousPaymentRequests)
+    expect(confirmedPaymentRequests[0].dueDate).toBe('09/11/2022')
+    expect(confirmedPaymentRequests[0].schedule).toBe('Q4')
+  })
+
+  test('should update schedule to cover remaining payments when quarterly and three remaining', () => {
+    const paymentRequests = [{
+      ledger: AP,
+      dueDate: '09/11/2022',
+      schedule: 'Q4',
+      value: -100
+    }]
+    const previousPaymentRequests = [{
+      paymentRequestNumber: 1,
+      ledger: AP,
+      dueDate: moment(new Date()).subtract(1, 'month').format('DD/MM/YYYY'),
```

Comment:
I see what you're doing with making this in the past by subtracting 1 month but then you're expecting new due date to be 2 months from now which is unintuitive given the schedule is quarterly.

I'm wondering if there's a more intuitive way of doing so -- defining a `mock past due date` for each might work but adds a bit of duplicated code

I dnt think it's a major problem and could be left